### PR TITLE
R4 Observation Labs and Vitals Writes

### PR DESCRIPTION
**Pull request by Manjushavrk**
Description
----
Added Observation Creates for Laboratory and Vital Signs in millennium R4 entities. 

PR Checklist
----
Screen Shot 2020-09-03 at 11 32 40 AM
Screen Shot 2020-09-03 at 11 32 49 AM
Screen Shot 2020-09-03 at 11 32 59 AM
Screen Shot 2020-09-03 at 11 33 09 AM
Screen Shot 2020-09-03 at 11 33 21 AM
Screen Shot 2020-09-03 at 11 33 31 AM
Screen Shot 2020-09-03 at 11 34 44 AM
Screen Shot 2020-09-03 at 11 34 59 AM
Screen Shot 2020-09-03 at 11 35 34 AM
Screen Shot 2020-09-03 at 11 35 42 AM
Screen Shot 2020-09-03 at 11 35 50 AM
Screen Shot 2020-09-03 at 11 35 58 AM
Screen Shot 2020-09-03 at 11 36 08 AM
Screen Shot 2020-09-03 at 11 36 17 AM
Screen Shot 2020-09-03 at 11 36 26 AM
Screen Shot 2020-09-03 at 11 36 36 AM
Screen Shot 2020-09-03 at 11 36 45 AM

- [x] Screenshot(s) of changes attached before changes merged.
- [ ] Screenshot(s) of changes attached after changes merged and published.

### CODE REVIEW

**Review by DanielWhiteCerner — APPROVED** [review]

**Review by fpivonka — APPROVED** [review]

File: lib/resources/r4/observation.yaml [hunk 1]
```diff
@@ -2,22 +2,79 @@
...
+              "display": 'Vital Signs'
+            }
+          ],
+          "text": 'Vital Signs'
+        }
+      ]
+    }
+  note: |
+    <ul>
+      <li> `Vital Signs` and `Laboratory` categories are supported. </li>
```
Comment (BS062721) [inline lib/resources/r4/observation.yaml:64]: should these be the actual value set codes?
`vital-signs` and `laboratory`

File: lib/resources/r4/types.yaml [hunk 2]
```diff
@@ -28,7 +28,9 @@ Reference: https://hl7.org/fhir/R4/references.html
 Patient: https://hl7.org/fhir/R4/patient.html
 Practitioner: https://hl7.org/fhir/R4/practitioner.html
 Period: https://hl7.org/fhir/R4/datatypes.html#Period
+performerFunction: https://hl7.org/fhir/R4/extension-event-performerfunction.html?
```
Comment (BS062721) [inline lib/resources/r4/types.yaml:31]: is that an extra `?` in the end?

**Review by BS062721 — APPROVED** [review]

File: lib/resources/r4/observation.yaml (hunk 1 above)
Comment (BS062721) [reply lib/resources/r4/observation.yaml:64]: maybe Fran wants to make this call. I am fine either way

File: lib/resources/r4/types.yaml (hunk 2 above)
Comment (Manjushavrk) [reply lib/resources/r4/types.yaml]: removed : https://github.com/cerner/fhir.cerner.com/commit/034df7e326ddbeda3b8fed3d80802b5e3a36b1ab#diff-b3c25dd9e50fb8a137ff641afe13c3b7L31

File: lib/resources/r4/observation.yaml [hunk 3]
```diff
@@ -30,22 +87,216 @@ fields:
...
+    <ul>
+      <li> Laboratory support `valueQuantity`, `valueCodeableConcept` and `valueString` </li>
+    </ul>
+  url: https://hl7.org/fhir/R4/observation-definitions.html#Observation.value[x]
 
 - name: dataAbsentReason
   required: 'No'
   type: CodeableConcept
   description: Provides a reason why the expected value in the element Observation.value[x] is missing.
+  action: terminolgy
```
Comment (amcginn) [inline lib/resources/r4/observation.yaml:274]: terminology?

File: lib/resources/r4/observation.yaml [hunk 4]
```diff
@@ -55,6 +306,170 @@ fields:
...
+            ],
+            "text": 'Normal Range'
+          }
+        }
+      ]
+    }
+  note: |
+    <ul>
+      <li> Only one referenceRange is supported </li>
+      <li> ReferenceRange.text is not supported when referenceRange.high and/or referenceRange.low are provided </li>
```
Comment (amcginn) [inline lib/resources/r4/observation.yaml:403]: field names should be quoted as in other note sections.

File: lib/resources/r4/observation.yaml [hunk 5]
```diff
@@ -55,6 +306,170 @@ fields:
...
+            "code": 'normal',
+            "display": 'Normal Range'
+          }
+        ],
+        "text": 'Normal Range'
+      }
+    }
+  note: |
+    <ul>
+      <li> Only a ReferenceRange.type of normal is supported </li>
```
Comment (amcginn) [inline lib/resources/r4/observation.yaml:470]: quoted/code formatted

**Review by amcginn — APPROVED** [review]

File: lib/resources/r4/observation.yaml [hunk 6]
```diff
@@ -30,22 +87,216 @@ fields:
...
+    <ul>
+      <li> Must be a reference to Practitioner. </li>
+      <li> Vital Signs require performer and supports up to two performers. </li>
+    </ul>
+  url: https://hl7.org/fhir/R4/observation-definitions.html#Observation.performer
+
+- name: performer.extension
+  required: 'No'
+  type: performerFunction
+  description:  Distinguishes the type of involvement of the performer in the event.
```
Comment (BS062721) [inline lib/resources/r4/observation.yaml:201]: is there extra space before `Distinguishes`?

**Review by BS062721 — APPROVED** [review]

File: lib/resources/r4/observation.yaml (hunk 6 above)
Comment (Manjushavrk) [reply lib/resources/r4/observation.yaml]: removed extra space : https://github.com/cerner/fhir.cerner.com/commit/a3c496078c9f8365f3c8e5b8e0e7d18a4d98b3ae#diff-53f5b7c6241dd914461705ddc9523c7eL201

File: lib/resources/r4/observation.yaml (hunk 5 above)
Comment (Manjushavrk) [reply lib/resources/r4/observation.yaml]: updated: https://github.com/cerner/fhir.cerner.com/commit/a3c496078c9f8365f3c8e5b8e0e7d18a4d98b3ae#diff-53f5b7c6241dd914461705ddc9523c7eR469

File: lib/resources/r4/observation.yaml (hunk 4 above)
Comment (Manjushavrk) [reply lib/resources/r4/observation.yaml]: updated: https://github.com/cerner/fhir.cerner.com/commit/a3c496078c9f8365f3c8e5b8e0e7d18a4d98b3ae#diff-53f5b7c6241dd914461705ddc9523c7eR402

File: lib/resources/r4/observation.yaml (hunk 3 above)
Comment (Manjushavrk) [reply lib/resources/r4/observation.yaml]: removed: https://github.com/cerner/fhir.cerner.com/commit/a3c496078c9f8365f3c8e5b8e0e7d18a4d98b3ae#diff-53f5b7c6241dd914461705ddc9523c7eL274

**Review by DanielWhiteCerner — APPROVED** [review]

**Comment by MaxPhilips** [comment]
verified on live

Screen Shot 2020-09-03 at 5 54 16 PM

**Pull request merged**